_data-prepper/common-use-cases/pattern-matching.md

```diff
@@ -0,0 +1,215 @@
+---
+layout: default
+title: Grok pattern matching with Data Prepper
+parent: Common use cases
+nav_order: 35
+---
+
+# Grok pattern matching with Data Prepper
+
+Data Prepper provides pattern matching capabilities with the [Grok processor]({{site.url}}{{site.baseurl}}/data-prepper/pipelines/configuration/processors/grok/). The Grok processor is based on the [`java-grok`](https://mvnrepository.com/artifact/io.krakens/java-grok) library and supports all compatible patterns. The `java-grok` library is built using the [`java.util.regex`](https://docs.oracle.com/javase/8/docs/api/java/util/regex/package-summary.html) regular expression library.
+
+You can add custom patterns to your pipelines using the `patterns_definitions` option. When debugging custom patterns, the [Grok Debugger](https://grokdebugger.com/) can be helpful.
+
+## Basic usage
+
+To get started with pattern matching, create the following pipeline:
+
+```json
```

Comment:
@lizsnyder Please confirm this modified pipeline is accurate for Data Prepper. 

Original from OpenSearch Service:

```
version: "2"
patten-matching-pipeline:
  source
    ...
  processor:
    - grok:
        match:
          message: ['%{IPORHOST:clientip} \[%{HTTPDATE:timestamp}\] %{NUMBER:response_status:int}']
  sink:
    - opensearch:
        # Provide an OpenSearch Service domain endpoint
        # Enable the 'serverless' flag if the sink is an OpenSearch Serverless collection
        aws:
          ...
        index: "metrics_for_traces"
        # serverless: true

```

Reply:
So anytime my docs discuss an "OpenSearch Service domain endpoint", you would want to replace with "OpenSearch cluster endpoint".

We call clusters "domains".